storage/blockchain/src/types.rs

```diff
@@ -187,7 +189,7 @@ pub struct BlockInfo {
     /// The adjusted block size, in bytes.
     ///
     /// See [`block_weight`](https://monero-book.cuprate.org/consensus_rules/blocks/weights.html#blocks-weight).
-    pub weight: u64,
+    pub weight: usize,
```

Comment:
Is there a reason things are mostly `usize` instead of `u64`? Here but also across the codebase in general, e.g. DB service messages.

They are required to be `u64` elsewhere (especially in RPC) and these `usize`s permeate and cause `usize_to_u64()` to be everywhere. I understand indices but these aren't that. Can we start using `u64` instead?

Reply:
It's a trade of between what type is needed more, I used to use `u64`s everywhere but I changed it here: #201 and I do remember there being an improvement.

This is a weight which is often calculated by getting the size of a Vec, which is given in a `usize`. Plus you could change the JSON types to be `usize`s right?

Reply:
Changing this actually removed multiple conversions in the DB

Reply:
The types are defined as 64-bit in `monerod`, I don't think defining public fields within public RPC types as `usize` is a good tradeoff. Probably won't affect many machines but not worth it IMO. I'm okay with using the cast functions if these need to be `usize` elsewhere.